crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -352,6 +352,12 @@ module sui::staking_pool {
             if (total_sui_withdraw_amount >= principal_withdraw_amount)
                 total_sui_withdraw_amount - principal_withdraw_amount
             else 0;
+        // This may happen when we are withdrawing everything from the pool and
+        // the rewards pool balance may be 1 less than reward_withdraw_amount.
```

Comment:
It doesn't necessarily have to be 1 less though.

Reply:
right although I've only seen 1 in test output, we don't know if it can only be 1. changing the comments